### PR TITLE
Bump all modules to latest

### PR DESCRIPTION
**Pull request by eed3si9n**
This integrates latest modules and existing ripple PRs.

### CODE REVIEW

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

**Comment by typesafe-tools** [comment]
A dbuild validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/99/console">is in progress...</a>

**Comment by typesafe-tools** [comment]
[//]: # (Refs: pull/3322/head-1.0-1.0-1.0-1.0-1.0.x)
dbuild <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/99/">has checked</A> the following projects against Scala 2.12:

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/pull/3322">pull/3322/head      </A> | sbt/sbt@f0ac6ae11cf472e6f6e1b294daa84df0a2fbaa23
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/1.0">1.0                 </A> | sbt/zinc@daaec42de7db636084384545015dbceeab51c7b6
  io                | <A HREF="https://github.com/sbt/io/tree/1.0">1.0                 </A> | sbt/io@49a9cea2906ecb62ea690b5d5dddc257e0ada151
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/tree/1.0">1.0                 </A> | sbt/librarymanagement@0147e0c77578ca4852137a6113bcac4c7adeeb38
  util              | <A HREF="https://github.com/sbt/util/tree/1.0">1.0                 </A> | sbt/util@5b85002a69dfcae4429fb6034831f4766aba509b
  website           | <A HREF="https://github.com/sbt/website/tree/1.0.x">1.0.x               </A> | sbt/website@1721355e3823bc3c83dac88a2c07f7727b2f9d26

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild?sbt=pull/3322/head&zinc=1.0&io=1.0&librarymanagement=1.0&util=1.0&website=1.0.x&deploy_to_bintray=false">(restart)</A>

**Comment by eed3si9n** [comment]
Ran scripted over night:

[error] (sbtRoot/*:scripted) Failed tests:
- [x] [error] 	actions / external-doc
- [x]  [error] 	actions / cross
- [x]  [error] 	compiler-project / separate-analysis-per-scala
- [x]  [error] 	compiler-project / error-in-invalidated
- [x]  [error] 	compiler-project / macro-config
- [x]  [error] 	compiler-project / semantic-errors
- [x]  [error] 	dependency-management / update-sbt-classifiers
- [x]  [error] 	dependency-management / cached-resolution-conflicts
- [x]  [error] 	dependency-management / pom-advanced
- [x]  [error] 	dependency-management / make-pom
- [x]  [error] 	dependency-management / cache-update
- [x]  [error] 	dependency-management / extra
- [x]  [error] 	dependency-management / cached-resolution-classifier
- [x]  [error] 	dependency-management / exclude-dependencies
- [x]  [error] 	dependency-management / pom-scope
- [x]  [error] 	project / auto-plugins
- [x]  [error] 	project / extra
- [x]  [error] 	source-dependencies / transitive-memberRef
- [x]  [error] 	source-dependencies / trait-member-modified
- [x]  [error] 	tests / scala-instance-classloader
- [x]  [error] 	tests / serial

[error] Total time: 7287 s, completed Jul 16, 2017 3:22:24 AM

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

File: sbt/src/sbt-test/compiler-project/macro-config/build.sbt [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
+ivyConfigurations += Macro
 
 // add the compiler as a dependency for src/macro/
 libraryDependencies +=
-  scalaVersion("org.scala-lang" % "scala-compiler" % _ % "macro").value
+  scalaVersion("org.scala-lang" % "scala-compiler" % _ % Macro).value
 
 // adds standard compile, console, package tasks for src/macro/
-inConfig(config("macro"))(Defaults.configSettings)
+inConfig(Macro)(Defaults.configSettings)
```
Comment (jvican) [inline sbt/src/sbt-test/compiler-project/macro-config/build.sbt]: Much better.

File: notes/1.0.0.markdown [hunk 2]
```diff
@@ -38,10 +37,15 @@ This is the RC-1 release of sbt 1.0.
...
 - Changes `sourceArtifactTypes` and `docArtifactTypes` from `Set[String]` to `Seq[String]` settings.
-- Renames `ivyScala: IvyScala` to `scalaModuleInfo: ScalaModuleInfo`.
 
 The Scala Center is working with Lightbend to provide [an automatic migration tool][sbt-migration-rewrites].
 
+#### Zinc API changes
+
+- Java classes under the `xsbti.compile` package such as `IncOptions` hides the constructor. Use the factory method `xsbti.compile.Foo.of(...)`.
+- Renames `ivyScala: IvyScala` to `scalaModuleInfo: ScalaModuleInfo`.
+- `xsbti.Reporter#log(...)` takes `xsbti.Problem` as the parameter. Call `log(problem.position, problem.message, problem.severity)` to delegate to the older `log(...)`.
```
Comment (jvican) [inline notes/1.0.0.markdown]: Good catch. I'll open a ticket in Zinc to ask contributors to migrate all these tests to Zinc when possible.

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

**Comment by typesafe-tools** [comment]
A dbuild validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/100/console">is in progress...</a>

**Comment by typesafe-tools** [comment]
[//]: # (Refs: pull/3322/head-1.0-1.0-1.0-1.0-1.0.x)
dbuild <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/100/">has checked</A> the following projects against Scala 2.12:

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/pull/3322">pull/3322/head      </A> | sbt/sbt@96b5c3bb45f4ee7147fa118fa670e8a727f019fe
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/1.0">1.0                 </A> | sbt/zinc@daaec42de7db636084384545015dbceeab51c7b6
  io                | <A HREF="https://github.com/sbt/io/tree/1.0">1.0                 </A> | sbt/io@49a9cea2906ecb62ea690b5d5dddc257e0ada151
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/tree/1.0">1.0                 </A> | sbt/librarymanagement@0147e0c77578ca4852137a6113bcac4c7adeeb38
  util              | <A HREF="https://github.com/sbt/util/tree/1.0">1.0                 </A> | sbt/util@5b85002a69dfcae4429fb6034831f4766aba509b
  website           | <A HREF="https://github.com/sbt/website/tree/1.0.x">1.0.x               </A> | sbt/website@1721355e3823bc3c83dac88a2c07f7727b2f9d26

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild?sbt=pull/3322/head&zinc=1.0&io=1.0&librarymanagement=1.0&util=1.0&website=1.0.x&deploy_to_bintray=false">(restart)</A>

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

File: sbt/src/main/scala/Import.scala [hunk 3]
```diff
@@ -8,6 +8,8 @@ trait Import {
   type URI = java.net.URI
   type URL = java.net.URL
 
+  implicit def Seq2Vector[T](s: Seq[T]): Vector[T] = s.toVector
```
Comment (dwijnand) [inline sbt/src/main/scala/Import.scala:11]: Can someone explain to me the rational for this?

**Pull request merged**